# HG changeset patch
# User David Bateman <dbateman@free.fr>
# Date 1206723074 14400
#      Fri Mar 28 12:51:14 2008 -0400
# Node ID 0dff8d9bf2297a731a04ce6ae2144f4876f9426b
# Parent  c00d467bef6b69863fdac6adcdbbae68f2d28019
Fix for plot(zeros(1,0),zeros(1,0))

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,13 @@
+2008-03-27  David Bateman  <dbateman@free.fr>
+
+	* plot/__plt2__.m: Test if args are empty first so that
+	plot(zeros(1,0),zeros(1,0)) works as expected.
+
 2008-03-27  Bill Denney  <bill@denney.ws>
 
 	* time/datenum.m: Allow vector inputs in any orientation.
 
 	* strings/validatestring.m: New function.
 	* strings/Makefile.in (SOURCES): Add it to the list.
 
 	* general/nargoutchk.m: New function.
diff --git a/scripts/plot/__plt2__.m b/scripts/plot/__plt2__.m
--- a/scripts/plot/__plt2__.m
+++ b/scripts/plot/__plt2__.m
@@ -43,17 +43,19 @@ function retval = __plt2__ (h, x1, x2, o
     x1 = real (x1);
   endif
 
   if (any (any (imag (x2))))
     x2 = real (x2);
   endif
 
   h_set = false;
-  if (isscalar (x1))
+  if (isempty (x1) && isempty (x2))
+    retval = zeros (0, 1);
+  elseif (isscalar (x1))
     if (isscalar (x2))
       retval = __plt2ss__ (h, x1, x2, options, properties);
     elseif (isvector (x2))
       retval = __plt2sv__ (h, x1, x2, options, properties);
     else
       error ("__plt2__: invalid data for plotting");
     endif
   elseif (isvector (x1))
@@ -69,15 +71,13 @@ function retval = __plt2__ (h, x1, x2, o
   elseif (ismatrix (x1))
     if (isvector (x2))
       retval = __plt2mv__ (h, x1, x2, options, properties);
     elseif (ismatrix (x2))
       retval = __plt2mm__ (h, x1, x2, options, properties);
     else
       error ("__plt2__: invalid data for plotting");
     endif
-  elseif (isempty (x1) && isempty (x2))
-    retval = zeros (0, 1);
   else
     error ("__plt2__: invalid data for plotting");
   endif
 
 endfunction
